grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -69,6 +69,12 @@ class PermissionService {
 		return id != null && hasPermission(userish, resource, op)
 	}
 
+	boolean checkAnonymousAccess(Object resource, Operation op) {
```

Comment:
I think this is the same as `check(null, resource, op)` - you probably want to use that instead of implementing a new method, as there are some details which are now overlooked (permission expiration for example).